### PR TITLE
Add cats-effect metrics

### PR DESCRIPTION
**Pull request by janstenpickle**
Adds cats-effect MBean extraction via a callback. This inspects `ComputePoolSampler` and `LocalQueueSampler` for their integer value attributes and adds them to a metrics collection. `LocalQueueSampler` metrics will include a label indicating the queue number.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -20,7 +20,25 @@ val Scala213 = "2.13.10"
...
-lazy val root = tlCrossRootProject.aggregate(trace4Cats)
+lazy val root = tlCrossRootProject.aggregate(catsEffect, trace4Cats)
+
+lazy val catsEffect = project
+  .in(file("cats-effect"))
+  .settings(
+    name := "prometheus4cats-contrib-cats-effect",
+    libraryDependencies ++= Seq(
+      "com.permutive" %% "prometheus4cats" % "1.0.0-RC3",
+      "org.typelevel" %% "cats-effect" % "3.3.14",
```
Comment (TimWSpence) [inline build.sbt:31]: Should these dependencies be extracted somewhere to enforce that each module depends on the same version?

**Review by TimWSpence — APPROVED** [review]
Nice!

File: build.sbt (hunk 1 above)
Comment (janstenpickle) [reply build.sbt:31]: Ah yeah, prom4cats probably should, but CE is only needed on this module

**Review by TimWSpence — APPROVED** [review]
LGTM!

**Pull request merged**